include/net/http_client.h

```diff
@@ -1,33 +1,20 @@
+/** @file
+ * @brief HTTP client high-level api implementation for Zephyr.
+ */
+
 /*
  * Copyright (c) 2019 Unisoc Technologies INC.
  *
  * SPDX-License-Identifier: Apache-2.0
  */
 
-/**
- * @file
- *
- * @brief HTTP client high-level api implementation for Zephyr.
- */
-
-#ifndef __ZEPHYR_HTTP_CLIENT_H__
-#define __ZEPHYR_HTTP_CLIENT_H__
+#ifndef ZEPHYR_INCLUDE_HTTP_CLIENT_H_
+#define ZEPHYR_INCLUDE_HTTP_CLIENT_H_
 
 #include <net/http_parser.h>
 
-#ifdef __cplusplus
-extern "C" {
-#endif
-
-/**
- * @brief Http client library
- * @defgroup http client Library
- * @ingroup networking
- * @{
- */
-
 #ifdef CONFIG_HTTPS
-	#define CONFIG_NET_SOCKETS_SOCKOPT_TLS
+#define CONFIG_NET_SOCKETS_SOCKOPT_TLS
```

Comment:
config NET_SOCKETS_SOCKOPT_TLS                                                                                                                                                                          
      bool "Enable TCP TLS socket option support [EXPERIMENTAL]"
      select TLS_CREDENTIALS
     help
        Enable TLS socket option support which automatically establishes
        a TLS connection to the remote host.

It's not a good way. You should enable it in your proj.conf or Kconfig dependency.

Reply:
And, I have no idea about EXPERIMENTAL.